pkg/registry/storage/csistoragecapacity/strategy.go

```diff
@@ -74,7 +74,12 @@ func (csiStorageCapacityStrategy) PrepareForUpdate(ctx context.Context, obj, old
 func (csiStorageCapacityStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	newCSIStorageCapacityObj := obj.(*storage.CSIStorageCapacity)
 	oldCSIStorageCapacityObj := old.(*storage.CSIStorageCapacity)
-	errorList := validation.ValidateCSIStorageCapacity(newCSIStorageCapacityObj)
+	allowValidateLabelValueInLabelSelector := true
+	tempErrs := validation.ValidateCSIStorageCapacity(oldCSIStorageCapacityObj, true)
+	if len(tempErrs) > 0 {
+		allowValidateLabelValueInLabelSelector = false
```

Comment:
we only want to set this based on the current state of selector validation in the existing object, not any validation error of any type (comment applies to all places we're computing whether to validate selectors strictly based on the existing object)